### ISSUE TITLE
[Bug] 

### ISSUE THREAD

**Issue by WilliamWatterson86** [xamarin/Xamarin.Forms#14193]
### Description
Email.ComposeAsync crashes app with Java.Lang.IllegalStateException: FragmentContainer does not have a view

In my code I have a button that will open the email client with a preselected "To" address.  

This is the code I am using.
var message = new EmailMessage
                {
                    To = recipients,
                };
                await Email.ComposeAsync(message);

The call to Email.ComposeAsync looks like it working fine, I get the dialog to choose which email app I want to use, then when I select one, my app in the background crashes with this error.  This does not happen with other Xamarin Essentials things like "Share" options when email is selected.

### Steps to Reproduce

1. Call await Email.ComposeAsync(message);
2. Select email app you want to use
3. App crashes

### Expected Behavior
App should remain where it was until the user has finished sending their email 

### Actual Behavior
App crashes when email app is selected

### Basic Information

- Version with issue:
- Last known good version: Not sure
- IDE: VS2019
- Platform Target Frameworks: 
  - iOS:  
  - Android:  
  - UWP:   
- Android Support Library Version: AndroidX
- Nuget Packages: 
Xamarin.Forms 5.0.0.2012
Xamarin.Essentials: 1.6.1

- Affected Devices:

### Screenshots

![image](https://user-images.githubusercontent.com/32571925/110762415-ad332b00-8248-11eb-8658-c4e933f3c2ff.png)

JniEnvironment+InstanceMethods.CallNonvirtualVoidMethod (Java.Interop.JniObjectReference instance, Java.Interop.JniObjectReference type, Java.Interop.JniMethodInfo method, Java.Interop.JniArgumentValue* args)
JniPeerMembers+JniInstanceMethods.InvokeVirtualVoidMethod (System.String encodedMember, Java.Interop.IJavaPeerable self, Java.Interop.JniArgumentValue* parameters)
Activity.OnStart ()
FormsAppCompatActivity.OnStart () D:\a\1\s\Xamarin.Forms.Platform.Android\AppCompat\FormsAppCompatActivity.cs:417
Activity.n_OnStart (System.IntPtr jnienv, System.IntPtr native__this)
(wrapper dynamic-method) Android.Runtime.DynamicMethodNameCounter.23(intptr,intptr)
java.lang.IllegalStateException: Fragment FragmentContainer{2ab20f} (753ad94b-2740-4b04-8c2b-8b68181c275e) id=0x7} does not have a view
androidx.fragment.app.Fragment$4.onFindViewById Fragment.java:2888
androidx.fragment.app.FragmentStateManager.createView FragmentStateManager.java:501
androidx.fragment.app.FragmentStateManager.moveToExpectedState FragmentStateManager.java:282
androidx.fragment.app.FragmentStore.moveToExpectedState FragmentStore.java:112
androidx.fragment.app.FragmentManager.moveToState FragmentManager.java:1636
androidx.fragment.app.FragmentManager.dispatchStateChange FragmentManager.java:3112
androidx.fragment.app.FragmentManager.dispatchActivityCreated FragmentManager.java:3056
androidx.fragment.app.Fragment.performActivityCreated Fragment.java:2989
androidx.fragment.app.FragmentStateManager.activityCreated FragmentStateManager.java:577
androidx.fragment.app.FragmentStateManager.moveToExpectedState FragmentStateManager.java:285
androidx.fragment.app.FragmentStore.moveToExpectedState FragmentStore.java:112
androidx.fragment.app.FragmentManager.moveToState FragmentManager.java:1636
androidx.fragment.app.FragmentManager.dispatchStateChange FragmentManager.java:3112
androidx.fragment.app.FragmentManager.dispatchActivityCreated FragmentManager.java:3056
androidx.fragment.app.FragmentController.dispatchActivityCreated FragmentController.java:251
androidx.fragment.app.FragmentActivity.onStart FragmentActivity.java:473
androidx.appcompat.app.AppCompatActivity.onStart AppCompatActivity.java:210
crc643f46942d9dd1fff9.FormsAppCompatActivity.n_onStart(Native Method)
crc643f46942d9dd1fff9.FormsAppCompatActivity.onStart FormsAppCompatActivity.java:128
android.app.Instrumentation.callActivityOnStart Instrumentation.java:1433
android.app.Activity.performStart Activity.java:7980
android.app.ActivityThread.handleStartActivity ActivityThread.java:3578
android.app.servertransaction.TransactionExecutor.performLifecycleSequence TransactionExecutor.java:221
android.app.servertransaction.TransactionExecutor.cycleToPath TransactionExecutor.java:201
android.app.servertransaction.TransactionExecutor.executeLifecycleState TransactionExecutor.java:173
android.app.servertransaction.TransactionExecutor.execute TransactionExecutor.java:97
android.app.ActivityThread$H.handleMessage ActivityThread.java:2220
android.os.Handler.dispatchMessage Handler.java:107
android.os.Looper.loop Looper.java:237
android.app.ActivityThread.main ActivityThread.java:8016
java.lang.reflect.Method.invoke(Native Method)
com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run RuntimeInit.java:493
com.android.internal.os.ZygoteInit.main ZygoteInit.java:1076

**Comment by agusibrahim** [comment]
same here, i look report from appcenter and google play store, many my customer affected. help 

> androidx.fragment.app.Fragment$4.onFindViewById
> Java.Lang.IllegalStateException: Fragment FragmentContainer{21084c4} (e687202c-12b0-4c6d-a6b9-f87f17028f31) id=0x1} does not have a view

**Comment by gogolon** [comment]
I probably have the same problem:
```
java.lang.IllegalStateException: 
  at androidx.fragment.app.Fragment$4.onFindViewById (Fragment.java:2888)
  at androidx.fragment.app.FragmentStateManager.createView (FragmentStateManager.java:501)
  at androidx.fragment.app.FragmentStateManager.moveToExpectedState (FragmentStateManager.java:282)
  at androidx.fragment.app.FragmentStore.moveToExpectedState (FragmentStore.java:112)
  at androidx.fragment.app.FragmentManager.moveToState (FragmentManager.java:1636)
  at androidx.fragment.app.FragmentManager.dispatchStateChange (FragmentManager.java:3112)
  at androidx.fragment.app.FragmentManager.dispatchActivityCreated (FragmentManager.java:3056)
  at androidx.fragment.app.Fragment.performActivityCreated (Fragment.java:2989)
  at androidx.fragment.app.FragmentStateManager.activityCreated (FragmentStateManager.java:577)
  at androidx.fragment.app.FragmentStateManager.moveToExpectedState (FragmentStateManager.java:285)
  at androidx.fragment.app.FragmentStore.moveToExpectedState (FragmentStore.java:112)
  at androidx.fragment.app.FragmentManager.moveToState (FragmentManager.java:1636)
  at androidx.fragment.app.FragmentManager.dispatchStateChange (FragmentManager.java:3112)
  at androidx.fragment.app.FragmentManager.dispatchActivityCreated (FragmentManager.java:3056)
  at androidx.fragment.app.FragmentController.dispatchActivityCreated (FragmentController.java:251)
  at androidx.fragment.app.FragmentActivity.onStart (FragmentActivity.java:473)
  at androidx.appcompat.app.AppCompatActivity.onStart (AppCompatActivity.java:210)
  at crc643f46942d9dd1fff9.FormsAppCompatActivity.n_onStart (Native Method)
  at crc643f46942d9dd1fff9.FormsAppCompatActivity.onStart (FormsAppCompatActivity.java:128)
  at android.app.Instrumentation.callActivityOnStart (Instrumentation.java:1435)
  at android.app.Activity.performStart (Activity.java:8222)
  at android.app.ActivityThread.handleStartActivity (ActivityThread.java:3818)
  at android.app.servertransaction.TransactionExecutor.performLifecycleSequence (TransactionExecutor.java:221)
  at android.app.servertransaction.TransactionExecutor.cycleToPath (TransactionExecutor.java:201)
  at android.app.servertransaction.TransactionExecutor.executeLifecycleState (TransactionExecutor.java:173)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:97)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:2307)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:246)
  at android.app.ActivityThread.main (ActivityThread.java:8506)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:602)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:1130)
```

**Comment by jamesmontemagno** [comment]
Workaround is to just manually install the latest 'Xamarin.AndroidX.Fragment' nuget package.

**Comment by gogolon** [comment]
@jamesmontemagno Even after manually installing the mentioned package, the app keeps crashing on Android 11. What can I do?

**Comment by jamesmontemagno** [comment]
Did you install Xamarin.AndroidX.Fragment v1.2.5.4 ? or a newer one? looks like some investigation may need to be done.

**Comment by gogolon** [comment]
I've installed the latest version, 1.3.2. Should I roll back to v1.2.5.4? 
I don't even know what action in the app precedes the crashes, as I'm unable to reproduce them on emulators. I only get reports from the Play console, all the same as the one I've posted above.

**Comment by DevEddy** [comment]
Same here, any new infos on this?

**Comment by jamesmontemagno** [comment]
@gogolon earlier someone said rolling back to 1.2.5.4 fixed their issue.

**Comment by nielsbrg** [comment]
This error occurred for me whenever the app went from the background back into the foreground. OnResume was not even called, it just instantly crashed with this error. Rolled back to 1.2.5.4 only to get [this error](https://github.com/xamarin/Xamarin.Forms/issues/14042). I then installed Xamarin.AndroidX.Activity 1.2.2 and Fragment 1.2.5.4 to resolve everything.

**Comment by DevEddy** [comment]
I can confirm that the downgrade fixes the issue, thanks @nielsbrg 
I had to downgrade other dependencies as well. Hope this will get fixed soon.

**Comment by xleon** [comment]
Same thing here. Can't reproduce myself (debug or release version), but Google console is full of it.

I will try your workaround on my next release.

![image](https://user-images.githubusercontent.com/145087/117852913-b0449a80-b23c-11eb-88a6-25e64c05719a.png)

**Comment by xleon** [comment]
Confirmed that the workaround above fixed all of my crashes

![image](https://user-images.githubusercontent.com/145087/118280260-dfe3e480-b480-11eb-97ad-5e628f1e9ae7.png)

**Comment by alexanderdibenedetto** [comment]
Can confirm the issue still persists without downgrading to 1.2.5.4. Is there an estimate for fixing this?

**Comment by suggyd** [comment]
Xamarin.AndroidX.Fragment 1.2.5.4 seems to fix it, but that is a whole 6 versions behind current.  Is there any news on a fixed version?

**Comment by AntRemo** [comment]
Tested with 1.4.0 and still get same error... 

I am able to reproduce with: 
1. Launch App 
2. Tap Home
3. Run `adb shell "am kill <PackageName>"`
4. Relaunch app... KABOOM!

1.2.5.4 has no error.

FYI @DevEddy @jamesmontemagno

**Comment by EmilAlipiev** [comment]
This was i think all the time issue with Xamarin even on Xamarin.Android reported long time a go on SO. https://stackoverflow.com/questions/39068772/xamarin-forms-android-java-lang-illegalstateexception-fragment-does-not-have-a

It is crucial issue indeed. we need to find a solution on this.

**Comment by EmilAlipiev** [comment]
> d with 1.4.0 and still get same error...
> 
> I am able to reproduce with:

I am seeing the error in my appcenter logs but i am not able reproduce with the way you described even without having Xamarin.AndroidX.Fragment installed. I am testing on Android 11 device

**Comment by EmilAlipiev** [comment]
any idea how can reproduce and solve this? this is really getting annoying 100s of crashes

**Comment by pro777s** [comment]
> any idea how can reproduce and solve this? this is really getting annoying 100s of crashes

Did downgrading the library Xamarin.Android.Fragment to version 1.2.5.4 help you?

**Comment by suggyd** [comment]
@pro777s - yes it did for me

**Comment by EmilAlipiev** [comment]
yes but is that a good solution to go so many versions older? because when you try to downgrade Fragment package, it automatically downgrades all related packages. So you need to basically use all AndroidX packages from  1.2.5.4. I don't find this is a good solution.

**Comment by pro777s** [comment]
@EmilAlipiev, I agree with you. I don't understand why there is no reaction from the Xamarin developers.

**Comment by agusibrahim** [comment]
there is no Xamarin.Android.Fragment installed on my android project, should i new install Xamarin.Android.Fragment 1.2.5.4 can solved this issue?

**Comment by EmilAlipiev** [comment]
Normally you don't need it. I also don't have it but for workaround it's claimed that you need to install the older version which makes no sense. It's pretty old and I am afraid that it can cause other issues

**Comment by masiton** [comment]
Can confirm, installing Xamarin.Android.Fragment 1.2.5.4 fixed a crash in our XF app when changing between dark/light mode. Same exception was occurring and also newer versions of this dependency didn't work. Specifically version 1.2.5.4 must be used to avoid crashing.

**Comment by agusibrahim** [comment]
> Can confirm, installing Xamarin.Android.Fragment 1.2.5.4 fixed a crash in our XF app when changing between dark/light mode. Same exception was occurring and also newer versions of this dependency didn't work. Specifically version 1.2.5.4 must be used to avoid crashing.

which version of xamarin.forms you're used?

**Comment by masiton** [comment]
> which version of xamarin.forms you're used?

5.0.0.2012.

**Comment by Codelisk** [comment]
Any updates on this one? i would like to update :)

**Comment by Jakar510** [comment]
@Codelisk 

It's already has released.  See https://github.com/xamarin/Xamarin.Forms/releases/tag/release-5.0.0-sr4

**Comment by DevEddy** [comment]
Can anyone confirm that the SR fixes the issues?

**Comment by Jakar510** [comment]
@DevEddy 

I've been using it for a couple of weeks and have not had any issues.

**Comment by IeuanWalker** [comment]
Also noticed this issue in google develop console. It's one of the main crashes in out app -
![image](https://user-images.githubusercontent.com/6544051/127299142-8f476a27-6825-4bcd-9352-9ed7bf19dd40.png)

Only an issue with the latest version of our app, you can see the crashes come flooding in after the latest release - 
![image](https://user-images.githubusercontent.com/6544051/127299554-973d443a-9e1c-478f-bd86-d96ee7a3c4da.png)

There where a lot of change in that release (lots of accessibility work), but most notably was -
- upgrading Xamarin.Forms from `4.8.0.1560` to `5.0.0.2012`
- Replacing 'Plugin.Permissions' with xamarin.essentials version
- As stated above 'Xamarin.AndroidX.Fragment' was updated from `1.2.4.1` to `1.3.2`

**Comment by DevEddy** [comment]
@IeuanWalker have you tried Xamarin.Forms 5.0.0.2083? This seems to be the SR4 where the issue should be fixed.

**Comment by IeuanWalker** [comment]
@DevEddy no not yet.

I'll update it for the next release.

It's hard to test any changes as it seams to happen randomly throughout the app

**Comment by DevEddy** [comment]
@AntRemo was able to reproduce this issue, maybe you can try it on your side.

> Tested with 1.4.0 and still get same error...
> 
> I am able to reproduce with:
> 
> 1. Launch App
> 2. Tap Home
> 3. Run `adb shell "am kill <PackageName>"`
> 4. Resume app... KABOOM!
> 
> 1.2.5.4 has no error.
> 
> FYI @DevEddy @xleon @jamesmontemagno

**Comment by IeuanWalker** [comment]
Thanks, @DevEddy, missed that one.

Before updating the NuGet it was crashing following the steps above. And looking in the device log it was the same issue.
After updating the NuGet i couldn't get it crash following the same steps.

So fingers crossed the issue is solved 🤞


Will report back once we release a new version and see whether the number of crashes go down or not

**Comment by xleon** [comment]
@IeuanWalker  any results on this?

**Comment by FunbiOyede** [comment]
Yeah @IeuanWalker any positive results yet?

**Comment by IeuanWalker** [comment]
@xleon @FunbiOyede 

Not really :/

Our testers have 2 devices that can replicate the issue. The best we got was to update XF to the latest version. This prevents it from crashing, by restarting the app, so it is better than before. But ideal they should be returned to the original view.

But for now, that's the best we could do, due to other commitments.



Haven't published a new version to the stores yet, but crash reports should hopefully go down.

**Comment by FunbiOyede** [comment]
Ok. Thanks for your response

**Comment by xleon** [comment]
I had to update XF to latest due to this and other crashes in Android. 
Everything looks good on our local testing. I removed manual references to AndroidX packages, leaving the automatic dependencies from XF. 
Will go to production soon. Wish me luck 😁

**Comment by kcrg** [comment]
@xleon You can remove Plugin.CurrentActivity nuget because [Xamarin.Essentials](https://docs.microsoft.com/en-us/dotnet/api/xamarin.essentials.platform.currentactivity) have this functionality. Small tip from me :)

**Comment by xleon** [comment]
Just passing by to confirm that latest XF works fine ☑️ in production without version tricks on AndroidX libraries (check my csproj https://github.com/xamarin/Xamarin.Forms/issues/14193#issuecomment-906808249)

**Comment by IeuanWalker** [comment]
@xleon you might not be seeing any crashes but i wouldn't say this issue is fixed.

See my previous comment - 

>The best we got was to update XF to the latest version. This prevents it from crashing, by restarting the app, so it is better than before. But ideal they should be returned to the original view.

**Comment by angelru** [comment]
My app in google play crashes with android.runtime.JavaProxyThrowable
crc643f46942d9dd1fff9.FormsAppCompatActivity.n_onStart

I have to remove all manual AndroidX dependencies and leave XF ones ??

**Comment by jaysidri** [comment]
My project did not have AndroidX.Fragment or AndroidX.Activity explicitly referenced but I've got the latest version of Xamarin.AndroidX.Biometrics and that seems to be pulling in the latest Activity / Fragment components. 

I had to manually add lower versions of these 2 nugets to resolve this issue. I've described this in more detail here:
https://stackoverflow.com/a/71490715/170169

**Comment by tentom** [comment]
I fixed this issue by adding the AndroidX.Fragment package with version 1.5.1 manually to my project, I also updated a couple of other AndroidX packages to make them compatible with AndroidX.Fragment 1.5.1. 

Running on Xamarin.Forms 5.0.0.2515

This will give some warning of packages outside of constrains. But at least my app are not crashing anymore. 

BTW it seems to happen when the activity gets killed so to debug locally you can tell Android to not keep activities in the developer settings and navigate away from the your app and then back again as described https://github.com/xamarin/Xamarin.Forms/issues/13895#issuecomment-813110989

**Comment by EmilAlipiev** [comment]
is the latest fragment version works without issue with the latest XF version now?

**Comment by iliketaiko** [comment]
Is there any update on they item?
